urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/builder/IgnoreBuilder.java

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2009 Hao Xu
+ertranne@hotmail.com
+
+This file is part of EriLex.
+
+EriLex is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package erilex.builder;
+
+import erilex.Global;
+import erilex.Grammar;
+import erilex.Or;
+import erilex.Production;
+import erilex.Star;
+
+/**
+ *
+ * @author ertri
+ */
+public class IgnoreBuilder implements Builder<Object> {
+    private ExpressionBuilder<RuleBuilder, Object> exp;
+    private final Grammar grammar;
+
+    public IgnoreBuilder(Grammar g) {
+        this.grammar = g;
+    }
+
+    public ExpressionBuilder<RuleBuilder, Object> start() {
+        return exp = new ExpressionBuilder(this, grammar);
+    }
+
+    public Production build() {
+        return null;
+    }
+
+    @Override
+    public Object exp(Production p) {
+        if(grammar.ignore == null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 52 to != check before setting the field value on line 56 or verify if the check is needed at all.